catalog/plugin/catalog-plugin-resource-usage/src/test/java/ddf/catalog/plugin/resource/usage/TestResourceUsagePlugin.java

```diff
@@ -0,0 +1,202 @@
+import static org.hamcrest.Matchers.is;
+import static org.hamcrest.Matchers.notNullValue;
+import static org.hamcrest.Matchers.nullValue;
+import static org.junit.Assert.assertThat;
+import static org.mockito.Matchers.any;
+import static org.mockito.Matchers.anyLong;
+import static org.mockito.Matchers.anyString;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.never;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.when;
+
+import java.io.Serializable;
+import java.security.Principal;
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.UUID;
+
+import org.apache.shiro.authz.Permission;
+import org.apache.shiro.mgt.DefaultSecurityManager;
+import org.apache.shiro.mgt.SecurityManager;
+import org.apache.shiro.realm.AuthorizingRealm;
+import org.apache.shiro.realm.Realm;
+import org.apache.shiro.session.mgt.SimpleSession;
+import org.apache.shiro.subject.PrincipalCollection;
+import org.apache.shiro.subject.SimplePrincipalCollection;
+import org.apache.shiro.subject.support.DelegatingSubject;
+import org.codice.ddf.persistence.PersistenceException;
+import org.codice.ddf.persistence.attributes.AttributesStore;
+import org.junit.Before;
+import org.junit.Test;
+import org.mockito.ArgumentCaptor;
+
+import ddf.catalog.data.Metacard;
+import ddf.catalog.operation.ResourceRequest;
+import ddf.catalog.plugin.PluginExecutionException;
+import ddf.catalog.plugin.StopProcessingException;
+import ddf.security.SecurityConstants;
+import ddf.security.Subject;
+
+public class TestResourceUsagePlugin {
+
+    private ResourceUsagePlugin plugin;
+
+    private AttributesStore attributeStore;
+
+    private static final String TEST_USER = "testuser";
+
+    private static final String RESOURCE_SIZE = "100";
+
+    private Subject subject;
+
+    @Before
+    public void setup() {
+        attributeStore = mock(AttributesStore.class);
+
+        plugin = new ResourceUsagePlugin(attributeStore);
+    }
+
+    @Test
+    public void testNullInput() throws StopProcessingException, PluginExecutionException {
+        assertThat(plugin.process(null), is(nullValue()));
+    }
+
+    @Test
+    public void testValidResourceSize()
+            throws StopProcessingException, PluginExecutionException, PersistenceException {
+
+        ArgumentCaptor<String> usernameArg = ArgumentCaptor.forClass(String.class);
+        ArgumentCaptor<Long> dataSizeArg = ArgumentCaptor.forClass(Long.class);
+
+        ResourceRequest origRequest = getMockResourceRequest(RESOURCE_SIZE, TEST_USER);
+        ResourceRequest request = plugin.process(origRequest);
+
+        assertThat(request, notNullValue());
+        assertThat(request, is(equalTo(origRequest)));
+
+        verify(attributeStore).updateUserDataUsage(usernameArg.capture(), dataSizeArg.capture());
+
+        assertThat(usernameArg.getValue(), is(equalTo(TEST_USER)));
```

Comment:
`is(equalTo())` can be simplified to [`is()`](http://hamcrest.org/JavaHamcrest/javadoc/1.3/org/hamcrest/core/Is.html#is(T)).